core/src/main/java/io/grpc/Metadata.java

```diff
@@ -324,11 +339,96 @@ private void mergePathAndAuthority(Metadata other) {
     public String toString() {
       return "Headers(path=" + path
           + ",authority=" + authority
+          + ",timeout=" + timeout
           + ",metadata=" + super.toStringInternal() + ")";
     }
   }
 
   /**
+   * Container class for timeout information as represented in Headers.
+   */
+  public static class Timeout extends Metadata {
```

Comment:
Note that with using a `Metadata.Key`, modifying the Netty and OkHttp clients would no longer be necessary. So you would be able to remove the changes to okhttp/Headers.java and netty/Utils.java on line 143